src/test/java/com/ss/training/utopia/counter/dao/UserDaoTests.java

```diff
@@ -0,0 +1,38 @@
+package com.ss.training.utopia.counter.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.ss.training.utopia.counter.entity.User;
+
+/**
+ * @author Justin O'Brien
+ */
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class UserDaoTests {
+
+	@Autowired
+	private TestEntityManager testEntityManager;
+	@Autowired
+	private UserDao userDao;
+
+	@Test
+	public void findByUsernameTest() {
+		String thisUsername = "ThisUsername", otherUsername = "OtherUsername";
+		User savedUser = new User(null, thisUsername, null, null, null),
+				otherUser = new User(null, otherUsername, null, null, null), foundUser;
+		testEntityManager.persist(savedUser);
+		testEntityManager.persist(otherUser);
+		testEntityManager.flush();
```

Comment:
What are these three commands with the testEntityManager doing?

Reply:
`persist` designates entities to be saved in H2, and `flush` more or less commits those changes.